cpp/velox/memory/VeloxMemoryPool.cc

```diff
@@ -81,21 +93,26 @@ class VeloxMemoryAllocatorVariant {
   MemoryAllocator* gluten_alloc_;
 };
 
-// The code is originated from /velox/common/memory/Memory.h
-// Removed memory manager.
-template <typename Allocator = VeloxMemoryAllocatorVariant, uint16_t ALIGNMENT = kNoAlignment>
+//  The code is originated from /velox/common/memory/Memory.h
+//  Removed memory manager.
 class WrappedVeloxMemoryPool : public facebook::velox::memory::MemoryPool {
  public:
   // Should perhaps make this method private so that we only create node through
   // parent.
-  explicit WrappedVeloxMemoryPool(
+  WrappedVeloxMemoryPool(
+      facebook::velox::memory::MemoryManager& memoryManager,
       const std::string& name,
       std::shared_ptr<MemoryPool> parent,
-      std::shared_ptr<Allocator> allocator,
-      int64_t cap = kMaxMemory)
-      : MemoryPool{name, parent}, localMemoryUsage_{}, cap_{cap}, allocator_{allocator} {
-    VELOX_USER_CHECK_GT(cap, 0);
-  };
+      std::shared_ptr<VeloxMemoryAllocatorVariant> glutenAllocator,
+      const Options& options = Options{})
+      : MemoryPool{name, parent, options.alignment},
+        cap_{options.capacity},
+        memoryManager_{memoryManager},
+        localMemoryUsage_{},
+        allocator_{memoryManager_.getAllocator()},
+        glutenAllocator_{glutenAllocator} {
```

Comment:
Have you checked if it's properly enough to allow copying `memoryManager` and `glutenAllocator`  (and the allocator probably) here? Maybe wrapping with `unique_ptr`s can make things clearer more or less.

Reply:
Velox do not use unique_ptr or shared ptr to wrap "allocator_".
https://github.com/facebookincubator/velox/blob/main/velox/common/memory/Memory.h#L408
I think we can still wrap std::shared_ptr for glutenAllocator like before.

Reply:
Oh I just see glutenAllocator is already wrapped with shared_ptr.